### PR TITLE
websocket: fix race when disconnecting right after connecting

### PR DESCRIPTION
**Pull request by lebauce**

### CODE REVIEW

**Comment by lebauce** [comment]
run skydive-unit-tests

**Comment by lebauce** [comment]
run skydive-cdd-overview-tests

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by lebauce** [comment]
run skydive-k8s-tests

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-elasticsearch
run skydive-functional-tests-backend-orientdb

**Comment by lebauce** [comment]
run skydive-k8s-tests
run skydive-functional-tests-backend-elasticsearch
run skydive-functional-tests-backend-orientdb

**Comment by lebauce** [comment]
run skydive-k8s-tests

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-k8s-tests

File: websocket/server.go [hunk 1]
```diff
@@ -102,23 +104,17 @@ func (s *Server) serveMessages(w http.ResponseWriter, r *auth.AuthenticatedReque
 	}
 
 	// call the incomerHandler that will create the Speaker
-	c, err = s.incomerHandler(conn, r)
+	c, err = s.incomerHandler(conn, r, func(c *wsIncomingClient) (Speaker, error) { return c, nil })
```
Comment (hunchback) [inline websocket/server.go]: why do we need a promoter if it just returns the input value (plus error = nil).

**Comment by lebauce** [comment]
run skydive-k8s-tests
run skydive-functional-tests-backend-elasticsearch

File: websocket/server.go (hunk 1 above)
Comment (lebauce) [reply websocket/server.go]: There are 2 implementations of the promoter : one that indeed returns the input value, the other that returns a `StructSpeaker`

**Comment by lebauce** [comment]
run skydive-k8s-tests

**Pull request merged**